### PR TITLE
Profile layout

### PR DESCRIPTION
**Pull request by oleksandr-samoilenko-ew**
set profile screen, build dob widget with calendar, add params to date picker fieds, create gender selection widget, add image picker

[Screen_recording_20240501_150323.webm](https://github.com/extrawest/extrawest_ui_kit_mobile/assets/68539527/6b94d03a-aa4b-4f3a-ac5f-57ca83fa84fe)

[Screen_recording_20240501_151518.webm](https://github.com/extrawest/extrawest_ui_kit_mobile/assets/68539527/923f329e-d79a-4282-8449-0d8ee033415b)

### CODE REVIEW

File: example/lib/layouts/user_profile_layout.dart [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return UserProfile(
+      emailController: emailController,
+      userNameController: displayNameController,
+      address1Controller: address1Controller,
+      address2Controller: address2Controller,
+      fieldsLabelStyle: const TextStyle(
+        color: Color(0xFF4C4639),
```
Comment (andriy-gulak-ew) [inline example/lib/layouts/user_profile_layout.dart:33]: use theme extensions

File: example/lib/layouts/user_profile_layout.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+class UserProfileLayoutState extends State<UserProfileLayout> {
+  bool isLoginPressed = false;
+  bool showRequiredEmailText = false;
+  bool showRequiredDisplayNameText = false;
+  bool showRequiredAddress1NameText = false;
+  bool showRequiredAddress2NameText = false;
+  final emailController = TextEditingController();
+  final displayNameController = TextEditingController();
+  final address1Controller = TextEditingController();
+  final address2Controller = TextEditingController();
```
Comment (andriy-gulak-ew) [inline example/lib/layouts/user_profile_layout.dart:23]: make it late final and initialize in initState

File: lib/components/widgets/ew_avatar_widget.dart [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  State<EWAvatarWidget> createState() => _EWAvatarWidgetState();
+}
+
+class _EWAvatarWidgetState extends State<EWAvatarWidget> {
+  final ImagePicker _picker = ImagePicker();
+  List<XFile>? _mediaFileList;
+  final TextEditingController maxWidthController = TextEditingController();
+  final TextEditingController maxHeightController = TextEditingController();
+  final TextEditingController qualityController = TextEditingController();
+  final TextEditingController limitController = TextEditingController();
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_avatar_widget.dart:36]: move to initState

File: lib/components/widgets/ew_avatar_widget.dart [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    this.pickImageGalleryText = 'Gallery',
+  });
+
+  @override
+  State<EWAvatarWidget> createState() => _EWAvatarWidgetState();
+}
+
+class _EWAvatarWidgetState extends State<EWAvatarWidget> {
+  final ImagePicker _picker = ImagePicker();
+  List<XFile>? _mediaFileList;
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_avatar_widget.dart:32]: init with empty constructor, make not-nullable

File: lib/components/widgets/ew_avatar_widget.dart [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        SizedBox(
+            width: 120,
+            height: 120,
+            child: Semantics(
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_avatar_widget.dart:79]: remove semantics

File: lib/components/widgets/ew_avatar_widget.dart [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+                                            File(_mediaFileList![index].path),
+                                            width: 120,
+                                            height: 120,
+                                            fit: BoxFit.cover,
+                                          )
+                                        : const SizedBox()),
+                              );
+                            },
+                            itemCount: _mediaFileList!.length,
+                          )
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_avatar_widget.dart:107]: extract to local build widget

Comment (andriy-gulak-ew) [reply lib/components/widgets/ew_avatar_widget.dart:107]: refactor inner ternary operator

File: lib/components/widgets/ew_avatar_widget.dart [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+                ),
+              ),
+            )
+            // _previewImages(),
+            ),
+      ],
+    );
+  }
+
+  Positioned buildSelectImageButton(BuildContext context) {
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_avatar_widget.dart:120]: - refactor inkwell splash animation, 
- return Widget

File: lib/components/widgets/ew_avatar_widget.dart [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+                          _onImageButtonPressed(ImageSource.camera,
+                              context: context);
+                        }),
+                  ],
+                );
+              });
+        },
+        child: Container(
+          width: MediaQuery.of(context).size.width,
+          color: Colors.black.withOpacity(0.2),
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_avatar_widget.dart:155]: move to theme extentions

File: lib/components/widgets/ew_gender_checkbox_list.dart [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    this.onGenderChanged,
+    this.genderBlockTitle = 'Gender',
+  });
+
+  @override
+  EWGenderCheckboxListState createState() => EWGenderCheckboxListState();
+}
+
+class EWGenderCheckboxListState extends State<EWGenderCheckboxList> {
+  late String selectedIndex;
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_gender_checkbox_list.dart:22]: consider removing late

File: lib/components/widgets/ew_gender_checkbox_list.dart [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        Text(widget.genderBlockTitle),
+        Container(
+          padding: EdgeInsets.zero,
+          width: MediaQuery.of(context).size.width,
+          child: Wrap(
+            runAlignment: WrapAlignment.start,
+            crossAxisAlignment: WrapCrossAlignment.start,
+            alignment: WrapAlignment.start,
+            spacing: 0.0,
+            children: [
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_gender_checkbox_list.dart:44]: Review spacing

File: lib/components/widgets/ew_gender_checkbox_list.dart [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+              for (String gender in widget.genderList)
+                IntrinsicWidth(
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/ew_gender_checkbox_list.dart:46]: review IntrinsicWidth usage

File: lib/components/widgets/text_field_widgets/date_input.dart [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final DateFormat formatter = DateFormat(widget.dateFormat);
+
+    return GestureDetector(
+      onTap: () => _selectDate(context),
+      child: AbsorbPointer(
```
Comment (andriy-gulak-ew) [inline lib/components/widgets/text_field_widgets/date_input.dart:67]: review usage

File: lib/layouts/user_profile/user_profile.dart [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+    super.key,
+    this.emailController,
+    this.fieldsLabelStyle,
+    this.fieldsTextStyle,
+    this.fieldsStrutStyle,
+    this.eMailOnChanged,
+    this.eMailOnTap,
+    this.eMailPrefixIcon,
+    this.emailHintText = 'Email',
+    this.emailLabelText = 'Email',
```
Comment (andriy-gulak-ew) [inline lib/layouts/user_profile/user_profile.dart:71]: remove duplicated initialization

File: lib/layouts/user_profile/user_profile.dart [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+
+  @override
+  State<UserProfile> createState() => _UserProfileState();
+}
+
+class _UserProfileState extends State<UserProfile> {
+  @override
+  Widget build(BuildContext context) {
+    return LayoutWrapper(
+      appBar: EWAppBar.medium(
```
Comment (andriy-gulak-ew) [inline lib/layouts/user_profile/user_profile.dart:116]: remove

File: lib/layouts/user_profile/user_profile.dart [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+typedef OnPickImageCallback = void Function(
+    double? maxWidth, double? maxHeight, int? quality, int? limit);
```
Comment (andriy-gulak-ew) [inline lib/layouts/user_profile/user_profile.dart:228]: move to the top of the file

**Review by andriy-gulak-ew — COMMENTED** [review]
Let's double-check that we can't pick a video file

File: lib/components/widgets/text_field_widgets/date_input.dart (hunk 12 above)
Comment (oleksandr-samoilenko-ew) [reply lib/components/widgets/text_field_widgets/date_input.dart:67]: AbsorbPointer is required to show calendar modal

**Review by andriy-gulak-ew — APPROVED** [review]

**Pull request merged**